### PR TITLE
Rework the Import SPI to be configurable via the Config API

### LINKED ISSUE #17663: Rework the import SPI to handle imports for the new store
Similar to issue #13613 and PR #17311, the import SPI needs to receive the same changes to allow the new file store to run its imports.

### PR DESCRIPTION
**Pull request by ahus1**
Also rework the export/import CLI for Quarkus, so that runtime options are available.

Closes #17663

### CODE REVIEW

**Comment by ahus1** [comment]
@vramik / @pedroigor / @mabartos / @vmuzikar - This PR adds the support to add runtime options to the export and import commands as discussed earlier this week. 

To implement this, it picks up the idea from @mabartos that `Picocli` asks the commands what they support to avoid too many ifs in this class. It now does so by three new methods in `AbstractCommand`: `#includeRuntime()`, `#includeBuildTime()`  and `#getOptionCategories()`. As discussed on the call, the different commands then filter out the options that don't apply to them. 

The part to implement the automatic re-augmentation for export/import will be implemented in a different PR. 

This PR completes the work started in #17311 for the export to make the import SPI more generic, so that it will fit the upcoming functionality for the file store.

I'm now waiting for the build to complete and will then add @mabartos and @vramik as reviewers.

cc: @hmlnarik

**Comment by ahus1** [comment]
@mabartos / @vramik - this PR has now a green build and is ready for review. Thanks!

File: quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java
Comment (mabartos) [inline quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java:378]: Thinking about this statement. We'll in the near future also provide the possibility to have Export with an optimized flag and probably even more commands in the future with that; this will be kinda messy. 

I was thinking something like this[1]. It's just a quick prototype, but it probably would be good to consider. For all 'Dependent' options, we just check the default action, and that's mainly related to the `--optimized` flag. But in the future, we could provide method sth like `evaluateDependentOptions(List<String> cliArgs){...}`, where we could specify when the dependent option should include also `build` process. 

However, it'd need to be properly documented and as these flags are used only on one place ATM, I don't see the high need to provide the additional logic. 

We can discuss this in the future; kinda out of scope of this PR.

[1] https://github.com/mabartos/keycloak/commit/31ae219a33ce109142f46a655acccbd6a6ce40d0

**Review by mabartos — COMMENTED** [review]
@ahus1 LGTM! Thanks.

I've just added one nitpick related to the name of the variable. And one thing for consideration, but it's not a blocker for this PR at all :)

File: quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java
Comment (mabartos) [reply quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java:378]: @ahus1 Agree, thanks.

**Review by mabartos — DISMISSED** [review]
@ahus1 Changes look good to me for the Quarkus distribution part. Thanks

**Review by mabartos — APPROVED** [review]

**Review by vramik — APPROVED** [review]
LGTM, thanks @ahus1

**Comment by ahus1** [comment]
@mhajas / @vmuzikar - now it's a maintainers turn to approve this. Could you please have a look (and re-run the WebAuthn?). 

Thanks!

**Comment by ahus1** [comment]
@mhajas / @vmuzikar / @pedroigor - now it's a maintainers turn to approve this. Could you please have a look (and re-run the WebAuthn?). 

Thanks!

**Comment by ahus1** [comment]
Rerunning build as upstream has now fixed WebAuthn

**Comment by ahus1** [comment]
@mhajas @vmuzikar - the build is now green after the change in main, so it can be merged when you think it is ready.

**Comment by vmuzikar** [comment]
Thanks @ahus1! I plan to review tomorrow.

**Review by vmuzikar — APPROVED** [review]
@ahus1 Very nice work, I think it's a move in the right direction.

LGTM, I'd be happy to merge this. Do we need a store maintainer review (maybe @mhajas)? Also maybe @pedroigor would want to give it a look.

File: quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/command/Start.java
Comment (ahus1) [reply quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/command/Start.java:98]: @vmuzikar - I'd like to keep it for now, until I rework the auto-build functionality for export/import. Once I do that, I need to update that logic anyway. WDYT?

Comment (vmuzikar) [reply quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/command/Start.java:98]: Works for me, thanks.

**Comment by ahus1** [comment]
@mhajas - with the CND's ok, and @vramik's review, could you please approve and merge it? Thanks!

**Review by mhajas — APPROVED** [review]
Approving per @vramik 's approval

**Comment by mhajas** [comment]
@pedroigor Do you want to have a look before this is merged?

**Review by pedroigor — APPROVED** [review]
@ahus1 Not really nice to see dependencies between commands when filtering options.

But we can refactor later to find the best approach to delegating to commands the resolution of their option categories.

**Pull request merged**
